kafkautils/producer.go

```diff
@@ -305,22 +299,78 @@ retry:
 		}
 	}
 
+	err = p.BeginTransaction()
```

Comment:
why do you move it here?

Reply:
It's so I can call `WriteAndCommit()` in a loop. As far as I know, calling `BeginTransaction()` at the start or after did not have an effect on transaction timeouts.